theano/tensor/signal/pool.py

```diff
@@ -310,23 +366,25 @@ def perform(self, node, inp, out):
         elif self.mode != 'max':
             func = numpy.average
 
-        for n in xrange(x.shape[0]):
-            for k in xrange(x.shape[1]):
-                for r in xrange(pr):
-                    row_st = r * st0
-                    row_end = builtins.min(row_st + ds0, img_rows)
-                    if not inc_pad:
-                        row_st = builtins.max(row_st, self.padding[0])
-                        row_end = builtins.min(row_end, x.shape[-2] + pad_h)
-                    for c in xrange(pc):
-                        col_st = c * st1
-                        col_end = builtins.min(col_st + ds1, img_cols)
-                        if not inc_pad:
-                            col_st = builtins.max(col_st, self.padding[1])
-                            col_end = builtins.min(col_end,
-                                                   x.shape[-1] + pad_w)
-                        zz[n, k, r, c] = func(y[
-                            n, k, row_st:row_end, col_st:col_end])
+        # precompute the region boundaries for each dimension
```

Comment:
you seem to have optimized the python code. That is great. Just to let you know that it isn't needed as it is only used for debugging. We can keep the optimized code without problem.